src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -138,14 +138,75 @@ public static CanvasDocument LoadFromSource(string directory2, ErrorContainer er
             // Load template files, recreate References/templates.json
             LoadTemplateFiles(errors, app, Path.Combine(directory2, PackagesDir), out var templateDefaults);
 
-            foreach (var file in dir.EnumerateFiles(AssetsDir))
-            {
+            // The resource entries for sample data is sharded into individual json files.
+            // Add each of these entries back into Resrouces.json
+            var resources = new List<ResourceJson>();
+            
+            foreach (var file in dir.EnumerateFiles(AssetsDir, "*", false))
+            {
+                var fileEntry = file.ToFileEntry();
+                // although should never be the case, but if it happens that both Resources.json file and individual sharded files exist, then read both.
                 if (file._relativeName == "Resources.json")
                 {
                     app._resourcesJson = file.ToObject<ResourcesJson>();
-                    continue;
                 }
-                app.AddAssetFile(file.ToFileEntry());
+                else if(fileEntry.Name.GetExtension() == ".json")
+                {
+                    // if its a json file then this must be one of the sharded files from Resources.json
+                    resources.Add(file.ToObject<ResourceJson>());
+                }
+                else
+                {
+                    // if its not a json file then add it to the asset files.
+                    // eg. logo file.
+                    app.AddAssetFile(fileEntry);
+                }
+            }
+
+            if(resources.Count > 0)
+            {
+                // if there is no Resources.json file then app._resourcesJson would be null, so make sure to initialize it before adding resource entries.
+                if (app._resourcesJson == null)
+                {
+                    app._resourcesJson = new ResourcesJson();
+                }
+
+                // although we should not have resouces.json file when the individual sharded files are there.
+                // but if that happens then try to do a union of all the resources in Resources.json and the sharded files.
```

Comment:
If we don't expect resources.json, we should not have this logic and instead just bump the version